spark/dl/src/main/scala/com/intel/analytics/bigdl/dataset/MiniBatch.scala

```diff
@@ -95,11 +182,407 @@ class TensorMiniBatch[T](
   override def getTarget(): Activity = {
     target
   }
+
+  override def setValue(samples: Seq[Sample[T]])(implicit ev: TensorNumeric[T]): this.type = {
+    require(samples.length > 0, "samples is empty")
+    require(batchSize == 0 || samples.length <= batchSize, "setValue: samples's size doesn't " +
+      s"match mini batch size, excepted ${size()} got ${samples.length}")
+
+    if (featurePaddingParam.isDefined || labelPaddingParam.isDefined) {
+      val longestFeature = MiniBatch.findLongestFeatures(samples)
+      val longestLabel = MiniBatch.findLongestLabels(samples)
+
+      MiniBatch.arraySampleToMiniBatch[T](samples, this,
+        Some(longestFeature), Some(longestLabel),
+        featurePadding, featurePaddingStrategy, labelPadding, labelPaddingStrategy)
+    } else {
+      MiniBatch.arraySampleToMiniBatch[T](samples, this)
```

Comment:
val longestFeature = if(featurePaddingParam.isDefined) {
    Some(MiniBatch.findLongestFeatures(samples))
} else {
    None
}

val longestLabel = if(featurePaddingParam.isDefined) {
    Some(MiniBatch.findLongestLabels(samples))
} else {
    None
}

MiniBatch.arraySampleToMiniBatch[T](samples, this, longestFeature, longestLabel, featurePadding, featurePaddingStrategy, labelPadding, labelPaddingStrategy)